File: src/android/app/src/main/java/org/citra/citra_emu/features/cheats/model/CheatEngine.java
package org.citra.citra_emu.features.cheats.model;

import androidx.annotation.Keep;

public class CheatEngine {
    @Keep
    private final long mPointer;

    @Keep
    public CheatEngine(long titleId) {
        mPointer = initialize(titleId);
    }

    private static native long initialize(long titleId);

    @Override
    protected native void finalize();

    public native Cheat[] getCheats();

    public native void addCheat(Cheat cheat);

    public native void removeCheat(int index);

    public native void updateCheat(int index, Cheat newCheat);

    public native void saveCheatFile();
}

File: src/android/app/src/main/java/org/citra/citra_emu/features/cheats/model/CheatEngine.kt
// Copyright 2023 Citra Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.

package org.citra.citra_emu.features.cheats.model

import androidx.annotation.Keep

@Keep
object CheatEngine {
    external fun loadCheatFile(titleId: Long)
    external fun saveCheatFile(titleId: Long)

    external fun getCheats(): Array<Cheat>

    external fun addCheat(cheat: Cheat?)
    external fun removeCheat(index: Int)
    external fun updateCheat(index: Int, newCheat: Cheat?)
}
